### DISCUSSION TITLE
File extension change

### DISCUSSION THREAD

**Discussion by haddougui** [mdn/mdn-community#443] [Content] [answered]
Hi Community,

I created this issue hoping that someone in our Javascript community will help in changing a given file extension from a source one into a destination one.
So, I already make a research on chatgpt AI and other tools and found some results but it wasn't the required one.
The change has a drawback stating that the metadata of the file or its content it's erased.
The same when opening or changing the file extension on your desktop viewer.
So I was asked to export a bunch of files into a PDF extension. For example from a PNG extension into PDF....
In addition, it can be possible to do that task without reading the content.

Please feel free to view the issue and give your response as expected.

Here it is the source code:

**/**
   * export into user desktop attachments as pdf extension
   */
  exportToPdf(){
    this.attachments.forEach(item => {
        item.mimeType= AttachmentWebContentType.PDF;
        const fileName= item.filename.split('.')[0];
        item.filename= `${fileName}.pdf`;
    })
  }**

**Comment by bsmth** [comment, marked as answer]
Hi @haddougui - thanks for asking. Changing the file extension will not be enough to convert the file from a `.png` to a `.pdf`. It depends where you are running this program, but you can have a look at the following resources:

- https://imagemagick.org/script/command-line-tools.php - Imagemagick to convert images from the command line
- https://github.com/parallax/jsPDF jsPDF for PDF tools in a JavaScript library.

**Reply by haddougui** [reply]
Thanks for your response. Yes, it not will be enough but using extra external libraries will decrease your application performance

**Reply by haddougui** [reply]
There's a bunch of solutions like reading source file content then writing it into target file or reading content then erasing file then creating file with required content. Agree or you have another tip on the issue ?

**Reply by yin1999** [reply]
Hey @haddougui. I think creating pdf files is always a more complicated problem (if you want to implement it without relying on third-party libraries). Using the libraries suggested by @bsmth is a good start (why would we invent our own wheels, unless you think these wheels could not solve your problem, or you have a better idea).

If you really want to avoid relying on third-party libraries, referencing other people's work is also a good start.

**Reply by haddougui** [reply]
Hey @yin1999. Yes, thanks. The same opinion
